### PR TITLE
[Documentation] Various header documentation corrections and refinements.

### PR DESCRIPTION
**Pull request by ejensen**

### CODE REVIEW

**Comment by ejensen** [comment]
CI test failure (download timeout) appears unrelated to the change

File: AsyncDisplayKit/ASControlNode.mm [hunk 1]
```diff
@@ -63,7 +63,7 @@ @interface ASControlNode ()
   @abstract Enumerates the ASControlNode events included mask, invoking the block for each event.
   @param mask An ASControlNodeEvent mask.
   @param block The block to be invoked for each ASControlNodeEvent included in mask.
-  @param anEvent An even that is included in mask.
+    @param anEvent An even that is included in mask.
```
Comment (appleguy) [inline AsyncDisplayKit/ASControlNode.mm]: This change seems incorrect / revert indentation.

File: AsyncDisplayKit/Details/ASImageProtocols.h [hunk 2]
```diff
@@ -70,10 +70,10 @@ typedef NS_ENUM(NSUInteger, ASImageDownloaderPriority) {
  @param URL The URL of the image to download.
  @param callbackQueue The queue to call `downloadProgressBlock` and `completion` on.
  @param downloadProgress The block to be invoked when the download of `URL` progresses.
- @param progress The progress of the download, in the range of (0.0, 1.0), inclusive.
+   @param progress The progress of the download, in the range of (0.0, 1.0), inclusive.
```
Comment (appleguy) [inline AsyncDisplayKit/Details/ASImageProtocols.h]: Why are these indented, but not the others surrounding?

**Comment by appleguy** [comment]
@ejensen this is very nice of you, thanks!  Would love to merge, just making sure the indentations aren't errors.

File: AsyncDisplayKit/Details/ASImageProtocols.h (hunk 2 above)
Comment (ejensen) [inline AsyncDisplayKit/Details/ASImageProtocols.h]: The parameters are indented to differentiate the methods parameters from the parameters of parameters. Ex `progress` is a parameter of the `downloadProgress` parameter, not the method itself. I will flatten the indentation if that's prefered.

**Comment by aaronschubert0** [comment]
@ejensen I presume you found these through the latest Xcode 7.3 beta? There's also "errors" for depreciated methods that aren't addressed in this PR. Would be great to fix all in one shot.

**Comment by ejensen** [comment]
The indentation was flattened. There were places in `ASMultiplexImageNode` that used the hierarchical indentation before this PR. They have also been flattened.

Those 2 deprecated `ASImageCacheProtocol` methods now have the deprecated attribute.

**Comment by appleguy** [comment]
@ejensen I love this kind of diff. Thanks a lot for caring about this stuff!

Feel free to submit a change anytime, even if for a minor or single-line issue. I would also love to hear from you by email if you have or have questions or feedback about the framework.  (asyncdisplaykit@gmail.com)

**Pull request merged**